includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -3488,10 +3510,27 @@ public function update_order_status() {
 					}
 				}
 
+				$return_url = $this->get_return_url( $order );
+
+				if ( $is_changing_payment ) {
+					$payment_token = $this->get_payment_token( $order );
+					if ( class_exists( 'WC_Subscriptions_Change_Payment_Gateway' ) ) {
+						WC_Subscriptions_Change_Payment_Gateway::update_payment_method( $order, $payment_token->get_gateway_id() );
+						$notice = __( 'Payment method updated.', 'woocommerce-payments' );
+
+						if ( WC_Subscriptions_Change_Payment_Gateway::will_subscription_update_all_payment_methods( $order ) && WC_Subscriptions_Change_Payment_Gateway::update_all_payment_methods_from_subscription( $order, $token->get_gateway_id() ) ) {
+							$notice = __( 'Payment method updated for all your current subscriptions.', 'woocommerce-payments' );
+						}
+
+						wc_add_notice( $notice );
+					}
+					$return_url = method_exists( $order, 'get_view_order_url' ) ? $order->get_view_order_url() : $this->get_return_url( $order );
+				}
+
 				// Send back redirect URL in the successful case.
 				echo wp_json_encode(
 					[
-						'return_url' => $this->get_return_url( $order ),
+						'return_url' => $return_url,
```

Comment:
The default `get_return_url` redirects users to the order received page because subscription change nonce is gone at this point as we're in the 3DS confirmation follow up AJAX here, that's why I'm using `get_view_order_url`.